### PR TITLE
KOGITO-6886 Consolidate Event API between Data Index and Runtime

### PR DESCRIPTION
**Pull request by cristianonicolai**
Assembly:
- https://github.com/kiegroup/kogito-apps/pull/1277
- https://github.com/kiegroup/kogito-runtimes/pull/2090
- https://github.com/kiegroup/kogito-examples/pull/1191

### CODE REVIEW

**Review by nmirasch — APPROVED** [review]

File: data-index/data-index-service/data-index-service-common/pom.xml [hunk 1]
```diff
@@ -21,6 +21,10 @@
       <groupId>org.kie.kogito</groupId>
       <artifactId>kogito-events-api</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.kie.kogito</groupId>
+      <artifactId>kogito-services</artifactId>
```
Comment (tiagodolphine) [inline data-index/data-index-service/data-index-service-common/pom.xml]: I cant we move the event classes to some module just for API, I mean kogito-services contain a lot of stuff that is more related to the runtime itself. So probably we can try to extract all event classes to the kogito-events-api, or even kogito-api, wdyt?

Comment (cristianonicolai) [reply data-index/data-index-service/data-index-service-common/pom.xml]: done

**Comment by kie-ci1** [comment]
**(tests) - kogito-examples job** `#78` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 235
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.downstream.kogito-examples/78/testReport/org.kie.kogito.springboot.outbox/OutboxIT/testSendProcessEvents/">org.kie.kogito.springboot.outbox.OutboxIT.testSendProcessEvents</a></summary>
org.awaitility.core.ConditionTimeoutException: <br/>Assertion condition defined as a lambda expression in org.kie.kogito.springboot.outbox.OutboxIT 1 expectation failed.<br/>Expected status code <200> but was <404>.<br/> within 1 minutes.<br/>	at org.kie.kogito.springboot.outbox.OutboxIT.testSendProcessEvents(OutboxIT.java:129)<br/>Caused by: java.lang.AssertionError: <br/>1 expectation failed.<br/>Expected status code <200> but was <404>.<br/><br/>	at org.kie.kogito.springboot.outbox.OutboxIT.lambda$testSendProcessEvents$1(OutboxIT.java:136)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.downstream.kogito-examples/78/display/redirect

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-examples tests

**Review by tiagodolphine — APPROVED** [review]

**Pull request merged**